src/System.Management.Automation/DscSupport/JsonDscClassCache.cs

```diff
@@ -0,0 +1,3274 @@
+        /// Load DSC resources into Cache from moduleFolderPath.
+        /// </summary>
+        /// <param name="errors">Collection of any errors encountered during initialization.</param>
+        /// <param name="modulePathList">Module path from where DSC PS modules will be loaded.</param>
+        /// <param name="importInBoxResourcesImplicitly">
+        /// if module is inbox.
+        /// </param>
+        private static void LoadDSCResourceIntoCache(Collection<Exception> errors, List<string> modulePathList, bool importInBoxResourcesImplicitly)
+        {
+            foreach (string moduleDir in modulePathList)
+            {
+                if (!Directory.Exists(moduleDir)) continue;
+
+                var dscResourcesPath = Path.Combine(moduleDir, "DscResources");
+                if (Directory.Exists(dscResourcesPath))
+                {
+                    foreach (string resourceDir in Directory.EnumerateDirectories(dscResourcesPath))
+                    {
+                        IEnumerable<string> schemaFiles = Directory.EnumerateFiles(resourceDir, "*.schema.json");
+                        if (!schemaFiles.Any())
+                        {
+                            continue;
+                        }
+
+                        Tuple<string, Version> moduleInfo = GetModuleInfoHelper(moduleDir, importInBoxResourcesImplicitly, isPsProviderModule: false);
+                        if (moduleInfo == null)
+                        {
+                            continue;
+                        }
+
+                        foreach (string schemaFile in schemaFiles)
+                        {
+                            ImportClasses(schemaFile, moduleInfo, errors, importInBoxResourcesImplicitly);
+                        }
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Get the module name and module version.
+        /// </summary>
+        /// <param name="moduleFolderPath">
+        /// Path to the module folder
+        /// </param>
+        /// <param name="importInBoxResourcesImplicitly">
+        /// if module is inbox and we are importing resources implicitly
+        /// </param>
+        /// <param name="isPsProviderModule">
+        /// Indicate a internal DSC module
+        /// </param>
+        /// <returns></returns>
+        private static Tuple<string, Version> GetModuleInfoHelper(string moduleFolderPath, bool importInBoxResourcesImplicitly, bool isPsProviderModule)
+        {
+            string moduleName = "PsDesiredStateConfiguration";
+            if (!importInBoxResourcesImplicitly)
+            {
+                moduleName = Path.GetFileName(moduleFolderPath);
+            }
+
+            string manifestPath = Path.Combine(moduleFolderPath, moduleName + ".psd1");
+            s_tracer.WriteLine("DSC GetModuleVersion: Try retrieving module version information from file: {0}.", manifestPath);
+
+            if (!File.Exists(manifestPath))
```

Comment:
Windows only code?

Reply:
Discussed offline. The concern was case-sensitive path comparisons that these APIs are doing on Linux.
The mitigation is that this code is running on Windows only.